### ISSUE TITLE
Prevent connection timed out when requesting configs.php file on IIS in web.config

### ISSUE THREAD

**Issue by mattab** [matomo-org/matomo#15651]
Bug reported by customer about non working Heatmaps on IIS:

> I get an error when after installing the HeatMap plugin and creating my first recording.
It says:Requesting 'https://piwik.X/plugins/HeatmapSessionRecording/configs.php?idsite=1&trackerid=5lX6EM&url=http%3A%2F%2Ftest.test%2F' resulted in an error: curl_exec: connection timed out after 2016 milliseconds. hostname requested was: piwik.X. As a result, tracking Heatmaps and Session Recordings may not work. You may need to change your webserver configuration to allow access to this file via the Internet or Intranet.

## Solution

This modification will work for everyone who uses IIS.
The file I have modified is the web.config in the plugins-folder.
The only thing I've added is this part:
```
<alwaysAllowedUrls>
<add url="/plugins/HeatmapSessionRecording/configs.php" />
</alwaysAllowedUrls>
```
And the whole file with looks now like this after the changes:
```
<?xml version="1.0" encoding="UTF-8"?>

<configuration>

<system.webServer>

<security>

<requestFiltering>

<denyUrlSequences>

<add sequence=".php" />

</denyUrlSequences>

<alwaysAllowedUrls>

<add url="/plugins/HeatmapSessionRecording/configs.php" />

</alwaysAllowedUrls>

</requestFiltering>

</security>

</system.webServer>

</configuration> 
```

**Comment by mattab** [comment]
yes we need to change in core, so it just works by default.

**Comment by sgiehl** [comment]
I'm not familiar with IIS, but wouldn't it be possible to create a new `web.config` within the plugin instead maybe?

**Comment by tsteur** [comment]
It won't work when it is placed in the plugin folder @sgiehl . Because it's overwritten by the plugins folder basically

**Comment by sgiehl** [comment]
Should we add that directly to core. Or add a hook instead, that allows the plugin to do that?

**Comment by tsteur** [comment]
We can add it directly to core

**Comment by mattab** [comment]
This was fixed in  #15669
